cdap-distributions/src/hdinsight/pageblob-configure.sh

```diff
@@ -0,0 +1,142 @@
+    if [ -z "$CLUSTERNAME" ]; then
+        CLUSTERNAME=$(echo -e "import hdinsight_common.ClusterManifestParser as ClusterManifestParser\nprint ClusterManifestParser.parse_local_manifest().deployment.cluster_name" | python)
+        if [ $? -ne 0 ]; then
+            echo "[ERROR] Cannot determine cluster name. Exiting!"
+            exit 133
+        fi
+    fi
+    echo "Cluster Name=$CLUSTERNAME"
+}
+
+# Try to perform an Ambari API GET
+# Dies if credentials are bad
+__validateAmbariConnectivity() {
+    coreSiteContent=$(bash ${AMBARICONFIGS_SH} -u ${USERID} -p ${PASSWD} get ${ACTIVEAMBARIHOST} ${CLUSTERNAME} core-site)
+    if [[ ${coreSiteContent} == *"[ERROR]"* && ${coreSiteContent} == *"Bad credentials"* ]]; then
+        echo '[ERROR] Username and password are invalid. Exiting!'
+        exit 134
+    fi
+}
+
+# Given a config type, name, and value, set the configuration via the Ambari API
+# Dies if unsuccessful
+setAmbariConfig() {
+    local __configtype=${1} # core-site, etc
+    local __name=${2}
+    local __value=${3}
+    updateResult=$(bash ${AMBARICONFIGS_SH} -u ${USERID} -p ${PASSWD} set ${ACTIVEAMBARIHOST} ${CLUSTERNAME} ${__configtype} ${__name} ${__value})
+
+    if [[ ${updateResult} != *"Tag:version"* ]] && [[ ${updateResult} == *"[ERROR]"* ]]; then
+        die "[ERROR] Failed to update ${__configtype}: ${updateResult}"
+    fi
+}
+
+# Fetches value of fs.azure.page.blob.dir, appends '/cdap' to it, updates it via Ambari API, updates the local client configurations, and restarts cluster services
+updateFsAzurePageBlobDirForCDAP() {
+    currentValue=$(bash ${AMBARICONFIGS_SH} -u ${USERID} -p ${PASSWD} get ${ACTIVEAMBARIHOST} ${CLUSTERNAME} core-site | grep 'fs.azure.page.blob.dir' | cut -d' ' -f3 | sed -e 's/"\(.*\)"[,]*/\1/')
+    if [ -n ${currentValue} ]; then
+      if echo ${currentValue} | grep -q '/cdap'; then
+        echo "fs.azure.page.blob.dir already contains /cdap"
+        return 0
+      else
+        newValue="${currentValue},/cdap"
+      fi
+    else
+      newValue="/cdap"
+    fi
+    echo "Updating fs.azure.page.blob.dir to ${newValue}"
+    setAmbariConfig 'core-site' 'fs.azure.page.blob.dir' ${newValue} || die "Could not update Ambari config"
+    restartCdapDependentClusterServices
+}
+
+# Stop an Ambari cluster service
+stopServiceViaRest() {
+    if [ -z "${1}" ]; then
+        echo "Need service name to stop service"
+        exit 136
+    fi
+    SERVICENAME=${1}
+    echo "Stopping ${SERVICENAME}"
+    curl -u ${USERID}:${PASSWD} -i -H 'X-Requested-By: ambari' -X PUT -d '{"RequestInfo": {"context" :"Configure Azure page blob support for CDAP installation"}, "Body": {"ServiceInfo": {"state": "INSTALLED"}}}' http://${ACTIVEAMBARIHOST}:${AMBARIPORT}/api/v1/clusters/${CLUSTERNAME}/services/${SERVICENAME}
+}
+
+# Start an Ambari cluster service
+startServiceViaRest() {
+    if [ -z "${1}" ]; then
+        echo "Need service name to start service"
+        exit 136
+    fi
+    sleep 2
```

Comment:
I don't know... it's copied from microsoft examples.  Next PR i'm going to try to change up these API calls anyway and so will remove it there.